uni-pad/src/native_pad.rs

```diff
@@ -1,10 +1,46 @@
-// TODO
-pub fn gamepad_init() {}
+use gilrs::Gilrs;
+use gilrs::ev::{Axis, Button};
 
-pub fn gamepad_axis(_player_num: i32) -> (f32, f32) {
+static mut CTX: Option<Gilrs> = None;
+
+static BUTTON_TO_NECS: [Button; 4] = [Button::South, Button::East, Button::West, Button::North];
+
+pub fn gamepad_init() {
+    unsafe {
```

Comment:
You could use Once to make sure it only call once :
https://doc.rust-lang.org/beta/std/sync/struct.Once.html